### PR TITLE
Members public api - 76400118

### PR DESCRIPTION
**Pull request by sedyakin**
https://pyrus.com/t#id76400118

### CODE REVIEW

**Review by SBorzov — APPROVED** [review]

File: Requests/Builders/CreateMemberRequestBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Pyrus.ApiClient.Requests.Builders
+{
+   public  class CreateMemberRequestBuilder
+    {
+        private readonly CreateMemberRequest _request;
+
+        public CreateMemberRequestBuilder(string email)
```
Comment (mikheevnickita) [inline Requests/Builders/CreateMemberRequestBuilder.cs:13]: Здесь пробелы

File: Requests/Builders/CreateMemberRequestBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        {
+            _request = new CreateMemberRequest
+            {
+                Email = email
+            };
+        }
+
+		public CreateMemberRequestBuilder WithFirstName(string name)
```
Comment (mikheevnickita) [inline Requests/Builders/CreateMemberRequestBuilder.cs:21]: Тут табы

Comment (mikheevnickita) [inline Requests/Builders/CreateMemberRequestBuilder.cs:21]: The idea of Builder pattern is to provide humand readable method call chain to construct a request.
In this case all method have the same signature witch does not have a lot of value.

File: Requests/Builders/CreateMemberRequestBuilder.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Pyrus.ApiClient.Requests.Builders
+{
+   public  class CreateMemberRequestBuilder
+    {
+        private readonly CreateMemberRequest _request;
```
Comment (mikheevnickita) [inline Requests/Builders/CreateMemberRequestBuilder.cs:11]: This class have single costrcutor with only

File: Requests/Builders/CreateMemberRequestBuilder.cs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+			return this;
+		}
+
+		public CreateMemberRequestBuilder WithLastName(string name)
+		{
+			_request.LastName = name;
+			return this;
+		}
+
+		public CreateMemberRequestBuilder WithEmail(string email)
```
Comment (mikheevnickita) [inline Requests/Builders/CreateMemberRequestBuilder.cs:33]: This class have a single costrcutor with email parameter. Having another method to override constructor parameter will lead to a confusion

File: Requests/Builders/CreateMemberRequestBuilder.cs [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+		{
+			_request.Email = email;
+			return this;
+		}
+
+		public CreateMemberRequestBuilder WithDepartmentId(int departmentId)
```
Comment (mikheevnickita) [inline Requests/Builders/CreateMemberRequestBuilder.cs:39]: WorkingAt(int departmentId)

File: Requests/Builders/CreateMemberRequestBuilder.cs [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+			return this;
+		}
+
+		public CreateMemberRequestBuilder WithPosition(string position)
+		{
+			_request.Position = position;
+			return this;
+		}
+
+		public CreateMemberRequestBuilder WithPhone(string phone)
```
Comment (mikheevnickita) [inline Requests/Builders/CreateMemberRequestBuilder.cs:51]: WorkPhone(string phone)

File: Requests/Builders/CreateMemberRequestBuilder.cs (hunk 2 above)
Comment (mikheevnickita) [reply Requests/Builders/CreateMemberRequestBuilder.cs:21]:
```
RequestBuilder
    .CreateAccount(login: "someone@domain.com")
        .For("John", "Doe")
        .WorkingAt(itDepartmentId)
        .As("CTO")
        .WorkPhone("+1(800) 123 4567")
        .Skype("john.doe")
        .Process(apiClient)
```

File: Requests/Builders/CreateMemberRequestBuilder.cs [hunk 7]
```diff
@@ -18,43 +18,32 @@ public CreateMemberRequestBuilder(string email)
             };
         }
 
-		public CreateMemberRequestBuilder WithFirstName(string name)
+		public CreateMemberRequestBuilder For(string name, string lastName)
```
Comment (alexeyvashchenko) [inline Requests/Builders/CreateMemberRequestBuilder.cs:21]: назови параметр firstName

**Review by alexeyvashchenko — CHANGES_REQUESTED** [review]

**Review by alexeyvashchenko — APPROVED** [review]

**Review by mikheevnickita — APPROVED** [review]

**Pull request merged**